File: Android/app/src/main/java/fr/uge/structsure/bluetooth/cs108/java/TagsIndex.java
package fr.uge.structsure.bluetooth.cs108.java;

import androidx.annotation.Keep;
import androidx.annotation.NonNull;

public class TagsIndex implements Comparable<TagsIndex> {
    private String address;
    private int position;

    @Keep
    public TagsIndex(String address, int position) {
        this.address = address;
        this.position = position;
    }

    @Keep public String getAddress() {
        return address;
    }
    @Keep public int getPosition() {
        return position;
    }

    @Override
    public int compareTo(@NonNull TagsIndex tagsIndex) {
        return address.compareTo(tagsIndex.address);
    }
}

File: Android/app/src/main/java/fr/uge/structsure/bluetooth/cs108/TagsIndex.kt
package fr.uge.structsure.bluetooth.cs108

import androidx.annotation.Keep


class TagsIndex @Keep constructor(
    @get:Keep val address: String,
    @get:Keep val position: Int
) : Comparable<TagsIndex?> {
    override fun compareTo(other: TagsIndex?): Int {
        return address.compareTo(other!!.address)
    }
}
